app/uk/gov/hmrc/tariffclassificationfrontend/controllers/SuppressCaseController.scala

```diff
@@ -38,35 +38,45 @@ class SuppressCaseController @Inject()(verify: RequestActions,
-  private val form: Form[Boolean] = MandatoryBooleanForm.form("suppress_case")
+  private val form: Form[String] = AddNoteForm.form
 
   override protected val config: AppConfig = appConfig
   override protected val caseService: CasesService = casesService
 
   override protected def redirect: String => Call = routes.CaseController.applicationDetails
   override protected def isValidCase(c: Case)(implicit request: AuthenticatedRequest[_]): Boolean = c.status == NEW
 
-  private def showCase(reference: String, f: Form[Boolean])
-                      (implicit request: AuthenticatedCaseRequest[AnyContent]): Future[Result] = {
-    getCaseAndRenderView(reference, c => successful(views.html.suppress_case(c, f)))
-  }
-
   def getSuppressCase(reference: String): Action[AnyContent] = (verify.authenticated andThen verify.casePermissions(reference) andThen
     verify.mustHave(Permission.SUPPRESS_CASE)).async { implicit request =>
-    showCase(reference, form)
+    getCaseAndRenderView(reference, c => successful(views.html.suppress_case(c, form)))
   }
 
-  def postSuppressCase(reference: String): Action[AnyContent] = (verify.authenticated andThen verify.casePermissions(reference) andThen
-    verify.mustHave(Permission.SUPPRESS_CASE)).async { implicit request =>
-
-    form.bindFromRequest().fold(
-      errors => showCase(reference, errors),
-      {
-        case true => validateAndRedirect(casesService.suppressCase(_, request.operator).map(c => routes.SuppressCaseController.confirmSuppressCase(reference)))
-        case _ => validateAndRedirect(c => successful(routes.SuppressCaseController.showContactInformation(reference)))
-      }
-    )
+  def postSuppressCase(reference: String): Action[MultipartFormData[Files.TemporaryFile]] =
+    (verify.authenticated andThen verify.casePermissions(reference) andThen verify.mustHave(Permission.SUPPRESS_CASE))
+      .async(parse.multipartFormData) { implicit request: AuthenticatedCaseRequest[MultipartFormData[Files.TemporaryFile]] =>
 
+    request.body.file("email").filter(_.filename.nonEmpty).filter(_.contentType.isDefined) match {
+      case Some(file) =>
+        form.bindFromRequest().fold(
+          errors =>
+            getCaseAndRenderView(reference, c => successful(views.html.suppress_case(c, errors))),
+          note => {
+            val upload = FileUpload(file.ref, file.filename, file.contentType.get)
+            validateAndRedirect(casesService.suppressCase(_, upload, note, request.operator).map(c => routes.SuppressCaseController.confirmSuppressCase(c.reference)))
+          }
+        )
+      case None =>
+        form.bindFromRequest().fold(
+          errors => {
+            val formWithErrors = errors.withError("email", "You must upload an email")
+            getCaseAndRenderView(reference, c => successful(views.html.suppress_case(c, formWithErrors)))
+          },
+          note => {
+            val formWithErrors = form.fill(note).withError("email", "You must upload an email")
```

Comment:
You must add a note?

Reply:
Already done as part of the form validation, annoyingly it cant handle files so we have to do that separately

Reply:
Gotcha